fairroot/base/sim/FairGenericStack.cxx

```diff
@@ -121,9 +131,9 @@ void FairGenericStack::FastSimMoveParticleTo(Double_t xx,
     PushTrack(tobedone, parent, pdg, px, py, pz, en, xx, yy, zz, tt, polx, poly, polz, proc, ntr, weight, status, -1);
     fFSMovedIndex = GetListOfParticles()->GetEntries() - 1;
     Int_t trackID = TVirtualMC::GetMC()->GetStack()->GetCurrentTrackNumber();
-    fFSTrackIter = fFSTrackMap.find(trackID);   // check if this track is not already created by FastSimulation
-    if (fFSTrackIter != fFSTrackMap.end())      // indeed the track has been created by the FastSimulation mechanism
-        trackID = fFSTrackIter->second;         // use the ID of the original track
+    auto tmentry = fFSTrackMap.find(trackID);   // check if this track is not already created by FastSimulation
+    if (tmentry != fFSTrackMap.end())           // indeed the track has been created by the FastSimulation mechanism
+        trackID = tmentry->second;              // use the ID of the original track
     fFSTrackMap[ntr] = trackID;
```

Comment:
This looks like some oversight from when `fFSTrackMap` was moved to `FairGenericStack`, no? `FairStack::GetCurrentTrackNumber()` also implements the FS-map-lookup. Should we rather delete it here and move the FS-map-lookup to `FairGenericStack::GetCurrentTrackNumber()` and remove the override `FairStack::GetCurrentTrackNumber()`?

Reply:
I guess, a whole lot more `fFSTrackMap` related things should probably be moved from FairStack to FairGenericStack.

I guess, the first one to move over from FairStack to FairGenericStack would be `fCurrentTrack`.

TBH: The question is basically, whether FairGenericStack should already have a "a good default implementation" of many things, or whether it should just be an API that people have to implement. For the later we could provide some FairMostlyUsableStack with some "good default implementation".

I think, we should leave that question for Radek and Florian as well.

Reply:
The current code does the FS-map-lookup twice, does it not?

Reply:
I think, it does, yes!

But I don't understand this, really.

Reply:
I have refactored the trackmap lookup into a new member function. This and #1588 show the double applying of that member function quite well. I guess, we can then file a nice issue.